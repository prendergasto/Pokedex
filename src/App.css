body {
  background-color: #282c34;
  color: white;
  padding: 40px;
  font-family: Arial;
  text-align: center;
}

ul {
  list-style-type: none;
  /* border: 1px white solid; */
}

/* #displayContainer {
  display: inline;
} */

li {
  border: 1px white solid;
}


/* body li {
  border: 1px;
  border-color: white;
}

ul {
  display: hidden;
} */