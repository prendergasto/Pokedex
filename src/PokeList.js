import React, { Component } from 'react';
import PokeItem from './Poke-item.js';
import pokemon from './data.js'

export default class PokeList extends Component {
    render() {
        return(
            <div id="displayContainer">
                <ol>
                    {this.props.data.map(pokemon => <PokeItem data={pokemon} />)}
                </ol>
            </div>
        )
    }
}


// data={pokemon}