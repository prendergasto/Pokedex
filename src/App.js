import React, { Component } from 'react';
import './App.css';
import request from 'superagent'
import PokeList from './PokeList.js';
// import images from './data.js';
// import pokeData from './data.js';
// import pokemon from './data.js';



export default class App extends React.Component {
  state = {
    selected: null,
    data: null,
  }
  async componentDidMount() {
    const data = await request.get('https://alchemy-pokedex.herokuapp.com/api/pokedex')

    console.log(data.body);
    this.setState({ data: data.body.results })
  }
  render() {
    return (
      <div id="requiredContainer">
      { this.state.data
      ? <PokeList data={this.state.data}/>
      : 'loading content' 
      } 
      </div>

    )
    }
}

// data={pokeData}