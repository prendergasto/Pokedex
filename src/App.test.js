import React from 'react';
// import { render } from '@testing-library/react';
import App from './App';
import renderer from "react-test-renderer";
import PokeList from './PokeList';
import PokeItem from './Poke-item';
test('renders App.js correctly', () => {
  const tree = renderer
  .create(<App />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});


test('renders App.js correctly', () => {
  const tree = renderer
  .create(<PokeList />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});


test('renders App.js correctly', () => {
  const tree = renderer
  .create(<PokeItem />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});