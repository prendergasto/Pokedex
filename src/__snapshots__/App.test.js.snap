// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App.js correctly 1`] = `
<div
  id="requiredContainer"
>
  loading content
</div>
`;
